frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -500,18 +500,20 @@ export const insightLogic = kea<insightLogicType>([
                 }),
             },
         ],
-        showTimeoutMessage: [false, { setShowTimeoutMessage: (_, { showTimeoutMessage }) => showTimeoutMessage }],
+        showTimeoutMessage: [false, { insightTimedOut: (_, { timedOutQueryId }) => !!timedOutQueryId }],
+        timedOutQueryId: [null as string | null, { insightTimedOut: (_, { timedOutQueryId }) => timedOutQueryId }],
         maybeShowTimeoutMessage: [
             false,
             {
                 // Only show timeout message if timer is still running
-                setShowTimeoutMessage: (_, { showTimeoutMessage }) => showTimeoutMessage,
+                insightTimedOut: (_, { timedOutQueryId }) => !!timedOutQueryId,
                 endQuery: (_, { exception }) => !!exception && exception.status !== 500,
                 startQuery: () => false,
                 setActiveView: () => false,
             },
         ],
-        showErrorMessage: [false, { setShowErrorMessage: (_, { showErrorMessage }) => showErrorMessage }],
+        showErrorMessage: [false, { insightErrored: (_, { erroredQueryId }) => !!erroredQueryId }],
```

Comment:
was torn on whether to make the `showErrorMessage` and `showTimeoutMessage` hold the query id and have the consumer know to `!!value` instead of letting the consumer have to know to read a second value to get the query id

I decided that while this has more moving pieces each idea is present once and so it's clearer for the future but... 🤷

Reply:
I kinda vote for less moving pieces 😅 `insightLogic` already has a _lot_ of them